### PR TITLE
Feature/marks table page

### PR DESCRIPTION
**Pull request by OlegAndrus12**
+Services
+DTO
+Controller
+View
Data mocked in RatingService.cs. Looking for the way to include links into layout without styles confussions.

### CODE REVIEW

File: WebApplication/Views/Rating/Rating.cshtml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                           <th>@item.SubjectID</th>
+                           <th>@item.Mark</th>
+                       </tr>   
+                      }
+                  </tbody>
+               </table>
+      </div>
+      <script>
+         $(function() {
+             $('#dataTable').DataTable({
```
Comment (nikilark) [inline WebApplication/Views/Rating/Rating.cshtml]: scripts look better in file for scripts

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 2]
```diff
@@ -214,4 +214,3 @@
     </div>
     
 </body>
-</html>
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml:217]: ?

File: WebApplication/wwwroot/js/dataTableScript.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+$(function() {
+    $('#dataTable').DataTable({
+        "pageLength": 4,
```
Comment (nikilark) [inline WebApplication/wwwroot/js/dataTableScript.js:3]: the same in view file, isnt it?

File: WebApplication/Views/Rating/Rating.cshtml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    Layout = null;
+}
+
+<!DOCTYPE html>
+<html lang="en">
+   <head>
+      <title>Ratings page</title>
+      <meta charset="utf-8">
+      <meta name="viewport" content="width=device-width, initial-scale=1">
+      <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
```
Comment (nikilark) [inline WebApplication/Views/Rating/Rating.cshtml]: remove layout`s part from view

File: Model/Entities/Rating.cs [hunk 5]
```diff
@@ -1,9 +1,12 @@
-﻿namespace Model
+﻿using System;
+
+namespace Model
 {
 	public class Rating
 	{
 		public int Id { get; set; }
 		public uint Mark { get; set; }
+		public DateTime Time { get; set; }
```
Comment (nikilark) [inline Model/Entities/Rating.cs]: this should also be in database

File: WebApplication/Views/Rating/Rating.cshtml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      <link href="https://cdn.datatables.net/1.10.18/css/dataTables.bootstrap4.min.css" rel="stylesheet">
+      <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
+      <!-- Bootstrap core JavaScript-->
+      <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+      <!-- Page level plugin JavaScript-->
+      <script src="https://cdn.datatables.net/1.10.18/js/jquery.dataTables.min.js"></script>
+      <script src="https://cdn.datatables.net/1.10.18/js/dataTables.bootstrap4.min.js"></script>
+   </head>
+   <body>
+            <div class="container" style="margin-top: 100px; text-align: center;">
```
Comment (nikilark) [inline WebApplication/Views/Rating/Rating.cshtml]: cannot styles be in css file?

File: WebApplication/Views/Rating/Rating.cshtml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                  </thead>
+                  <tfoot>
+                      <tr>
+                        <th>Date</th>
+                        <th>Subject ID</th>
+                        <th>Mark</th>
+                      </tr>
+                  </tfoot>
+                  <tbody>
+                      @foreach (var item in Model)
```
Comment (nikilark) [inline WebApplication/Views/Rating/Rating.cshtml]: maybe replace "item" with "MarkModel" (for example, you can find better name)

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 2 above)
Comment (nikilark) [reply WebApplication/Views/Timetable/Timetable.cshtml:217]: Why do you removed closing "html" tag in page you weren't even working at

File: WebApplication/Views/Rating/Rating.cshtml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+@model IEnumerable<Core.DTO.Rating>
+
+@{
+ViewData["Title"] = "Ratings";
+}
+
+<div class="container" style="margin-top: 100px; text-align: center;">
```
Comment (nikilark) [inline WebApplication/Views/Rating/Rating.cshtml]: move styles to css file

File: WebApplication/Views/Shared/_Layout.cshtml [hunk 9]
```diff
@@ -6,11 +6,17 @@
     <title>@ViewData["Title"] - WebApplication</title>
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
     <link rel="stylesheet" href="~/css/site.css" />
+    <link href="https://cdn.datatables.net/1.10.18/css/dataTables.bootstrap4.min.css" rel="stylesheet">
+    <link rel="stylesheet" href="~/css/timetableStyle.css">
```
Comment (nikilark) [inline WebApplication/Views/Shared/_Layout.cshtml]: look on Roman`s approach

File: WebApplication/Views/Shared/_Layout.cshtml [hunk 10]
```diff
@@ -6,11 +6,17 @@
...
 </head>
 <body>
     @RenderBody()
-    <script src="~/lib/jquery/dist/jquery.min.js"></script>
+    <script type="text/javascript" src ="~/js/timetableScript.js"></script>
     <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
+    <script src="~/js/dataTableScript.js"></script>
```
Comment (nikilark) [inline WebApplication/Views/Shared/_Layout.cshtml]: same as with css

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 2 above)
Comment (arihilmir) [reply WebApplication/Views/Timetable/Timetable.cshtml:217]: Bcoz it is not needed here. body tag too.

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 11]
```diff
@@ -211,7 +216,12 @@
...
         </main>
-    </div>
-    
+    </div>   
 </body>
-</html>
+
+@section Scripts
+{
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Comment (arihilmir) [inline WebApplication/Views/Timetable/Timetable.cshtml]: jquery should be included in _Layout. This is part for **your** scripts.

File: WebApplication/Controllers/RatingController.cs [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+using System.Linq;
+using System;
+
+namespace WebApplication.Controllers
+{
+    public class RatingController: Controller
+    {
+        private readonly IMapper _mapper;
+        private readonly IRatingService _rating;
+        private readonly ILogger<RatingController> _logger;
```
Comment (arihilmir) [inline WebApplication/Controllers/RatingController.cs]: Do you log anything?

File: WebApplication/Controllers/RatingController.cs [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+        public RatingController(IMapper mapper, IRatingService rating, ILogger<RatingController> logger)
+        {
+            _mapper = mapper;
+            _rating = rating;
+            _logger = logger;
+        }
+        public IActionResult Index()
+        {
+            List<Core.DTO.Rating> ratingsList = _rating.LoadRatingAsync().Result;
+            return (View("Rating", ratingsList));
```
Comment (arihilmir) [inline WebApplication/Controllers/RatingController.cs]: Remove redundant parentheses.

File: WebApplication/Views/Rating/Rating.cshtml [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+            <th>@markModel.Time</th>
+         </tr>
+         }
+      </tbody>
+   </table>
+</div>
+
+@section Scripts
+{
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Comment (arihilmir) [inline WebApplication/Views/Rating/Rating.cshtml]: jQuery should be included in _Layout. `Script` section is used for **your** js-files.

File: Services/RatingService/RatingService.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+using System.Threading.Tasks;
+using Core.DTO;
+
+namespace Services.RatingService
+{
+    public class RatingService: IRatingService
+    {
+        public async Task<List<Rating>> LoadRatingAsync()
+        {
+            List<Rating> listRatings = new List<Rating>()
```
Comment (arihilmir) [inline Services/RatingService/RatingService.cs]: Just return the list.

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 16]
```diff
@@ -2,8 +2,13 @@
 @{
     ViewData["Title"] = "Timetable";
 }
-<script type="text/javascript" src ="~/js/timetableScript.js"></script>
-<link rel="stylesheet" href="~/css/timetableStyle.css">
+
+@section Styles
+{
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.14.0/css/mdb.min.css" rel="stylesheet">
```
Comment (arihilmir) [inline WebApplication/Views/Timetable/Timetable.cshtml:8]: Bootstrap has to be included in _Layout.

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 17]
```diff
@@ -211,7 +216,12 @@
                 </div>
             </div>
         </main>
-    </div>
-    
+    </div>   
 </body>
```
Comment (arihilmir) [inline WebApplication/Views/Timetable/Timetable.cshtml]: This is not the end of **body** section.

File: WebApplication/Controllers/RatingController.cs (hunk 12 above)
Comment (OlegAndrus12) [reply WebApplication/Controllers/RatingController.cs]: not yet

Comment (arihilmir) [reply WebApplication/Controllers/RatingController.cs]: So, remove it.

**Review by arihilmir — APPROVED** [review]

**Pull request merged**